components/brave_federated/data_store_service.cc

```diff
@@ -39,20 +78,31 @@ void DataStoreService::OnInitComplete(bool success) {
 }
 
 void DataStoreService::Init() {
+  // data_stores_.clear();
+
   auto callback = base::BindOnce(&DataStoreService::OnInitComplete,
                                  weak_factory_.GetWeakPtr());
-  ad_notification_timing_data_store_.Init(
-      kAdNotificationTaskId, kAdNotificationTaskName, kMaxNumberOfRecords,
-      kMaxRetentionDays, std::move(callback));
+  ad_timing_data_store_.Init(kAdNotificationTaskId, kAdNotificationTaskName,
+                             kMaxNumberOfRecords, kMaxRetentionDays,
+                             std::move(callback));
+
+  // data_stores_.emplace(kAdNotificationTaskName,
```

Comment:
@aseren this is what I originally intended for the `data_stores_` variable, however even if this part actually works, when I go then to retrieve a data store with the getter function below there's a crash when I call any function on that specific data store. This is only the case after I retrieve it through the getter, when `Init` is called above it works fine

Reply:
Seems like there is dereferencing of deleted variable, as it stores pointer to local variable: `ad_notification_timing_data_store`.

It is hard to find the issue without actual code. Could you please upload code that has the crash?

Reply:
uploded 👍